### DISCUSSION TITLE
Unable to use docs in combination with stories in subdirectories

### DISCUSSION THREAD

**Discussion by bkardol** [storybookjs/storybook#14136] [Help]
In our Storybook (v6.1+) we'd like to have the following structure:
- Control (Top level)
  - Usage (Story)
  - States (Directory)
    - State 1 (Story)
    - State 2 (Story)
    - State 3 (Story)
  - Playground (Story)

What I did:
I created a component-states.tsx file with the states:

```typescript
import React from 'react';
import { h } from '@stencil/core';

export const StateOne = () => <Component />;
export const StateTwo = () => <Component />;
export const StateThree = () => <Component />;
```

and an mdx file importing the stories:
```typescript
import { Meta, Story } from '@storybook/addon-docs/blocks';
import Usage from './component.usage.mdx';
import { StateOne, StateTwo, StateThree } from './component-states';

<Meta title="Components/Checkbox" />

# Checkbox, Radio & Toggle

<Usage />

## States

A checkbox, radio button and toggle button can have several states. These states can be programmatically applied by the application for specific reasons.

<Story name="States/StateOne" story={StateOne} />

<Story name="States/StateTwo" story={StateTwo} />

<Story name="States/StateThree" story={StateThree} />

## Playground

<Story name="Playground" story={Playground} />
```

The result:
![StoryDocSubdirectories](https://user-images.githubusercontent.com/8373502/109970397-487e4a80-7cf5-11eb-8078-e966fa099091.png)

Is there a way to keep the doc with all the stories but group the stories into subdirectories? So in the example above: group the 'States' stories together.

**Comment by shilman** [comment]
This isn't currently supported. If you want to achieve nav structure you described, I think you would need to use [single-story hoisting](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#single-story-hoisting):

- /Components/Checkbox/Usage exporting Usage (hoisted)
- /Components/Checkbox/States exporting { StateOne, StateTwo, StateThree } (un-hoisted)
- /Components/Checkbox/Playground exporting Playground (hoisted)
